### ISSUE TITLE
attributionControl broken when compacted and given optional customAttribution string

### ISSUE THREAD

**Issue by ruizdurazo** [mapbox/mapbox-gl-js#7440]
**mapbox-gl-js version**: 0.50.0

**browser**: Chrome

### Steps to Trigger Behavior

 1. Create a new map with `attributionControl` set to `false`.
 2. Create a `new mapboxgl.AttributionControl` given the `compact` option set to `true` and a `customAtribution` string specified.

### Code

```
mapboxgl.accessToken = "-----------"

map = new mapboxgl.Map({
  attributionControl: false,
  container: 'map',
  style: 'mapbox://styles/mapbox/streets-v9',
  center: [0, 0],
  zoom: 0
})

map.addControl(new mapboxgl.AttributionControl({
  compact: true,
  customAttribution: "Example string"
}))
```

![image](https://user-images.githubusercontent.com/33944811/47176480-50cbbb80-d2db-11e8-9c2a-804cf584d4f4.png)

### Expected Behavior
The `customAttribution` string would be compacted and hidden along with the default attributions.

### Actual Behavior
The `customAttribution` string seems to be completely separate from the default attributions.

**Comment by areichman** [comment]
I saw this issue when upgrading from 0.47 to 0.51. I'm setting a custom attribution in a source style:

```javascript
map.addSource({
  type: 'raster',
  attribution: 'My Data <a href="https://my.com" target="_blank" rel="noopener">© My Company</a>',
  ...
})
```

Using a string with an anchor tag like this, the non-anchor text still bleeds out like shown in the original screenshot. For now, I've moved everything into the anchor text and it works as expected.

**Comment by ryanhamley** [comment]
@areichman the bug you describe should be fixed by https://github.com/mapbox/mapbox-gl-js/pull/7608 which just merged to `master` and will be part of 0.52. I tested out the attribution in your example on the current `master` branch and it works as expected in both compact and normal attributions.
